packages/bitget-wallet/src/lib/bitget-wallet.ts

```diff
@@ -0,0 +1,284 @@
+    }
+
+    const error = new Error(
+      typeof res.error === "string" ? res.error : res.error.type
+    );
+
+    // Prevent signing out by throwing.
+    if (error.message === "User reject") {
+      throw error;
+    }
+
+    // Continue signing out but log the issue.
+    logger.log("Failed to sign out");
+    logger.error(error);
+  };
+
+  const setupEvents = () => {
+    _state.wallet.on("accountChanged", async (newAccountId) => {
+      logger.log("onAccountChange", newAccountId);
+      emitter.emit("signedOut", null);
+    });
+
+    _state.wallet.on("rpcChanged", async (rpc) => {
+      logger.log("onNetworkChange", rpc);
+
+      if (options.network.networkId !== rpc.networkId) {
+        await signOut();
+
+        emitter.emit("signedOut", null);
+        emitter.emit("networkChanged", { networkId: rpc.networkId });
+      }
+    });
+  };
+
+  const getAccounts = async (): Promise<Array<Account>> => {
+    const accountId = _state.wallet.getAccountId() || "";
+    if (!accountId) {
+      return [];
+    }
+
+    const publicKey = await _state.wallet.getPublicKey();
+
+    return [
+      {
+        accountId,
+        publicKey: publicKey ? publicKey.toString() : undefined,
+      },
+    ];
+  };
+
+  const transformTransactions = (
+    transactions: Array<Optional<Transaction, "signerId">>
+  ) => {
+    return transactions.map((transaction) => {
+      return {
+        receiverId: transaction.receiverId,
+        actions: transaction.actions,
+      };
+    });
+  };
+
+  if (_state.wallet && _state.wallet.isSignedIn()) {
+    setupEvents();
+  }
+
+  return {
+    async signIn({ contractId, methodNames }) {
+      const existingAccounts = await getAccounts();
+
+      if (existingAccounts.length) {
+        return existingAccounts;
+      }
+
+      const { accessKey, error } = await _state.wallet.requestSignIn({
+        contractId,
+        methodNames,
+      });
+
+      // debugger;
+      if (!accessKey || error) {
+        await signOut();
+        throw new Error(
+          (typeof error === "string" ? error : error.type) ||
+            "Failed to sign in"
+        );
+      }
+
+      setupEvents();
+
+      return await getAccounts();
+    },
+    signOut,
+    async getAccounts() {
+      return getAccounts();
+    },
+    async verifyOwner({ message }) {
+      logger.log("BitgetWallet:verifyOwner", { message });
+
+      const networkId = options.network.networkId || "";
+      const accountId = (await _state.wallet.getAccountId()) || "";
+      const pubKey = (await _state.wallet.getPublicKey()) || "";
+
+      const block = await provider.block({ finality: "final" });
+
+      const data = {
+        accountId,
+        message,
+        blockId: block.header.hash,
+        publicKey: pubKey,
+        keyType: "0",
+      };
+      const encoded = JSON.stringify(data);
+
+      const signed = await _state.wallet.signMessage(
+        encoded,
+        accountId,
+        networkId
+      );
```

Comment:
This here is a bit confusing the public API of a wallet for `signMessage` should be implemented based on [NEP-413](https://github.com/near/NEPs/blob/master/neps/nep-0413.md).

If you have a look at the Sender Wallet there the signMessage of the signer is used to sing the payload for verifyOwner: https://github.com/near/wallet-selector/blob/main/packages/sender/src/lib/sender.ts#L242

The support for `signMessage` based on NEP-413 should be added in a similar way like this one https://github.com/near/wallet-selector/pull/997